src/Textile/API/events.ts

```diff
@@ -0,0 +1,279 @@
+  return new EventSubscription(
+    () => threadRemovedListeners = threadRemovedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addAccountPeerAddedListener(listener: (peerId: string) => void) {
+  accountPeerAddedListeners.push(listener)
+  return new EventSubscription(
+    () => accountPeerAddedListeners = accountPeerAddedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addAccountPeerRemovedListener(listener: (peerId: string) => void) {
+  accountPeerRemovedListeners.push(listener)
+  return new EventSubscription(
+    () => accountPeerRemovedListeners = accountPeerRemovedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addQueryDoneListener(listener: (queryId: string) => void) {
+  queryDoneListeners.push(listener)
+  return new EventSubscription(
+    () => queryDoneListeners = queryDoneListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addQueryErrorListeners(listener: (queryId: string, error: string) => void) {
+  queryErrorListeners.push(listener)
+  return new EventSubscription(
+    () => queryErrorListeners = queryErrorListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addClientThreadQueryResultListeners(listener: (queryId: string, thread: IThread) => void) {
+  clientThreadQueryResultListeners.push(listener)
+  return new EventSubscription(
+    () => clientThreadQueryResultListeners = clientThreadQueryResultListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addContactQueryResultListeners(listener: (queryId: string, contact: IContact) => void) {
+  contactQueryResultListeners.push(listener)
+  return new EventSubscription(
+    () => contactQueryResultListeners = contactQueryResultListeners.filter((item) => item !== listener),
+  )
+}
+
+eventEmitter.addListener('NODE_STARTED', () => {
+  for (const listener of nodeStartedListeners) {
+    listener()
+  }
+})
+
+eventEmitter.addListener('NODE_FAILED_TO_START', (error) => {
+  for (const listener of nodeFailedToStartListeners) {
+    listener(error)
+  }
+})
+
+eventEmitter.addListener('NODE_STOPPED', () => {
+  for (const listener of nodeStoppedListeners) {
+    listener()
+  }
+})
+
+eventEmitter.addListener('NODE_FAILED_TO_STOP', (error) => {
+  for (const listener of nodeFailedToStopListeners) {
+    listener(error)
+  }
+})
+
+eventEmitter.addListener('NODE_ONLINE', () => {
+  for (const listener of nodeOnlineListeners) {
+    listener()
+  }
+})
+
+eventEmitter.addListener('WILL_STOP_NODE_IN_BACKGROUND_AFTER_DELAY', (seconds) => {
+  for (const listener of willStopNodeInBackgroundAfterDelayListeners) {
+    listener(seconds)
+  }
+})
+
+eventEmitter.addListener('CANCELED_PENDING_NODE_STOP', () => {
+  for (const listener of canceledPendingNodeStopListeners) {
+    listener()
+  }
+})
+
+eventEmitter.addListener('NOTIFICATION_RECEIVED', (base64) => {
+  const notification = Notification.decode(Buffer.from(base64, 'base64'))
```

Comment:
These sdk subscriptions to the RN bridge events use our knowledge of the data coming over the bridge to convert it to typed data to send to the end subscribers